0MI0800304/7.03/ConsoleApplication1/problem123.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <fstream>
+//problem 1
+void writeData() {
+	std::ofstream file("result.txt");
+	if (!file.is_open()) {
+		std::cout << "Error!" << std::endl;
+		return;
+	}
+	int a = 0, b = 0, c = 0;
+	std::cin >> a >> b >> c;
+	file << a + b + c << " " << a * b * c << std::endl;
+	file.close();
+
+}
+void readData() {
+	std::ifstream file("result.txt");
+	if (!file.is_open()) {
+		std::cout << "Error!" << std::endl;
+		return;
+	}
+	int a = 0, b = 0;
+	file >> a >> b;
+	std::cout << a - b;
+	file.close();
+	}
+//problem 2
+int charCount() {
+	std::ifstream file("myFile.txt");
+	int count = 0;
```

Comment:
the declaration of this variable should be done either before the creation of `file` or after the `if` where you're checking the result of the file opening. The idea is to group together those pieces of logic that do relate to one another - file stuff is one, counters - another.